src/super_gradients/training/samples/pose_estimation_sample.py

```diff
@@ -0,0 +1,158 @@
+import dataclasses
+import numpy as np
+import torch
+
+from typing import Optional, List, Union
+
+from super_gradients.training.datasets.data_formats.bbox_formats.xywh import xywh_to_xyxy, xyxy_to_xywh
+
+__all__ = ["PoseEstimationSample"]
+
+
+@dataclasses.dataclass
+class PoseEstimationSample:
+    """
+    :param image:              Associated image with a sample. Can be in [H,W,C] or [C,H,W] format
+    :param image_layout:       Layout of the image (HWC or CHW)
+    :param mask:               Target mask in [H,W] format
+    :param joints:             Target joints in [NumInstances, NumJoints, 3] format.
+                               Last dimension contains (x,y,visibility) for each joint.
+    :param areas:              (Optional) Numpy array of [N] shape with area of each instance.
+                               Note this is not a bbox area, but area of the object itself.
+                               One may use a heuristic `0.53 * box area` as object area approximation if this is not provided.
+    :param bboxes:             (Optional) Numpy array of [N,4] shape with bounding box of each instance (XYWH)
+    :param additional_samples: (Optional) List of additional samples for the same image.
+    :param is_crowd:           (Optional) Numpy array of [N] shape with is_crowd flag for each instance
+    """
+
+    __slots__ = ["image", "mask", "joints", "areas", "bboxes", "is_crowd", "additional_samples"]
+
+    image: Union[np.ndarray, torch.Tensor]
+    mask: Union[np.ndarray, torch.Tensor]
+    joints: np.ndarray
+    areas: Optional[np.ndarray]
+    bboxes: Optional[np.ndarray]
+    is_crowd: Optional[np.ndarray]
+    additional_samples: Optional[List["PoseEstimationSample"]]
+
+    @classmethod
+    def compute_area_of_joints_bounding_box(cls, joints) -> np.ndarray:
+        """
+        Compute area of a bounding box enclosing visible joints for each pose instance.
+        :param joints: np.ndarray of [Num Instances, Num Joints, 3] shape (x,y,visibility)
+        :return:       np.ndarray of [Num Instances] shape with box area of the visible joints
+                       (zero if all joints are not visible or only one joint is visible)
+        """
+        visible_joints = joints[:, :, 2] > 0
+        xmax = np.max(joints[:, :, 0], axis=-1, where=visible_joints, initial=joints[:, :, 0].min())
```

Comment:
It is needed. When computing the bounding box of visible joints with respect to mask (visible_joints) we have to pass initial value. One can pass some -1_000_000_000 but I though that it would be just a bit nicer not to make any assumptions how big/small values there can be and take the smallest available coordinate.